pkg/apis/logging/v1alpha1/types.go

```diff
@@ -60,11 +60,20 @@ type ElasticsearchSpec struct {
 	Storage      v1alpha1.ElasticsearchNodeStorageSource `json:"storage"`
 }
 
-// This is the struct that will contain information pertinent to Log collection (Fluentd)
+// This is the struct that will contain information pertinent to Log and event collection
 type CollectionSpec struct {
-	Type        string `json:"type"`
+	LogCollection   LogCollectionSpec   `json:"logCollection,omitempty"`
+	EventCollection EventCollectionSpec `json:"eventCollection,omitempty"`
+	Normalizer      NormalizerSpec      `json:"normalizer,omitempty"`
+}
+
+type LogCollectionSpec struct {
+	Type        LogCollectionType `json:"type"`
 	FluentdSpec `json:"fluentd,omitempty"`
```

Comment:
so for rsyslog, we'll have both the FluentdSpec and
```
 	RsyslogSpec `json:"rsyslog,omitempty"`
```
?

Reply:
Yes